src/main/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessor.java

```diff
@@ -0,0 +1,198 @@
+import org.sagebionetworks.bridge.workerPlatform.exceptions.WorkerException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+
+/**
+ * Worker used to download all participants into a CSV and email it to researchers.
+ */
+@Component("DownloadParticipantRosterWorker")
+public class DownloadParticipantRosterWorkerProcessor implements ThrowingConsumer<JsonNode> {
+    private static final Logger LOG = LoggerFactory.getLogger(DownloadParticipantRosterWorkerProcessor.class);
+
+    // If there are a lot of downloads, write log messages regularly so we know the worker is still running.
+    private static final int REPORTING_INTERVAL = 100;
+
+    static final int PAGE_SIZE = 100;
+    private static final long THREAD_SLEEP_INTERVAL = 1000L;
+
+    private BridgeHelper bridgeHelper;
+    private FileHelper fileHelper;
+    private SesHelper sesHelper;
+    private DynamoHelper dynamoHelper;
+    private ZipHelper zipHelper;
+
+    /** Helps call Bridge Server APIs */
+    @Autowired
+    public final void setBridgeHelper(BridgeHelper bridgeHelper) {
+        this.bridgeHelper = bridgeHelper;
+    }
+
+    /** Wrapper class around the file system. Used by unit tests to test the functionality without hitting the real file
+     * system.
+     */
+    @Autowired
+    public final void setFileHelper(FileHelper fileHelper) {
+        this.fileHelper = fileHelper;
+    }
+
+    /** SES helper, used to email the pre-signed URL to the requesting user. */
+    @Autowired
+    public final void setSesHelper(SesHelper sesHelper) {
+        this.sesHelper = sesHelper;
+    }
+
+    /** Dynamo helper*/
+    @Autowired
+    public final void setDynamoHelper(DynamoHelper dynamoHelper) {
+        this.dynamoHelper = dynamoHelper;
+    }
+
+    @Autowired
+    public final void setZipHelper(ZipHelper zipHelper) {
+        this.zipHelper = zipHelper;
+    }
+
+    /** Main entry point into Download Participant Roster Worker. */
+    @Override
+    public void accept(JsonNode jsonNode) throws Exception {
+
+        BridgeDownloadParticipantRosterRequest request;
+        try {
+            request = DefaultObjectMapper.INSTANCE.treeToValue(jsonNode, BridgeDownloadParticipantRosterRequest.class);
+        } catch (IOException e) {
+            throw new PollSqsWorkerBadRequestException("Error parsing request: " + e.getMessage(), e);
+        }
+
+        String userId = request.getUserId();
+        String appId = request.getAppId();
+        String password = request.getPassword();
+        LOG.info("Received request for userId=" + userId + ", app=" + appId + ", password=" + password);
```

Comment:
It's okay to put the userId and appId in the logs, but not the password. If the logs somehow get leaked, this password would allow someone to potentially steal the participant roster.